lib/src/local_working_copy.rs

```diff
@@ -763,8 +765,41 @@ fn mtime_from_metadata(metadata: &Metadata) -> MillisSinceEpoch {
     )
 }
 
+/// On Windows, checks if a path is a reparse point (junction or symlink).
+/// Junctions have is_dir()=true but is_symlink()=false, so we need this
+/// separate check to avoid recursing into them and hitting "Access denied".
+#[cfg(windows)]
+fn is_reparse_point(metadata: &Metadata) -> bool {
+    const FILE_ATTRIBUTE_REPARSE_POINT: u32 = 0x400;
+    (metadata.file_attributes() & FILE_ATTRIBUTE_REPARSE_POINT) != 0
+}
+
+/// On Windows, checks if a DirEntry is a junction (directory reparse point).
+#[cfg(windows)]
+fn is_dir_entry_junction(entry: &DirEntry) -> bool {
+    entry
+        .metadata()
+        .map(|m| m.file_type().is_dir() && is_reparse_point(&m))
+        .unwrap_or(false)
+}
+
+#[cfg(not(windows))]
+fn is_dir_entry_junction(_entry: &DirEntry) -> bool {
+    false
+}
+
 fn file_state(metadata: &Metadata) -> Option<FileState> {
     let metadata_file_type = metadata.file_type();
+    // On Windows, junctions (created by tools like pnpm) have is_dir()=true
+    // but is_symlink()=false. We need to detect them via the reparse point
+    // attribute to avoid "Access denied" errors when trying to read their contents.
+    // On Windows, junctions (created by tools like pnpm) have is_dir()=true
+    // but is_symlink()=false. They cannot be read as symlinks or regular files.
+    // Returning None causes jj to skip them as "special files".
+    #[cfg(windows)]
+    if metadata_file_type.is_dir() && is_reparse_point(metadata) {
+        return None;
+    }
     let file_type = if metadata_file_type.is_dir() {
         None
```

Comment:
Perhaps, this wouldn't be needed because directories (and other special files) are mapped to `None`.